masterfabric-mobile/swift-camp
# CL - NavigationView Viper Implementation

## 📋 PR Description

This PR is an example created to demonstrate the use of NavigationView. Users can navigate to the detail page using NavigationLink, and a settings button is placed in the top toolbar. This provides a simple example to explain navigation functionality in SwiftUI.



---

## ✅ Checklist


---

## 🛠 Steps to Test


1. Launch the application.
2. Navigate to the NavigationView screen from HomeView by clicking the "NavigationView" button.
3. Test the back button to ensure it navigates correctly to the HomeView.


---

## 🔗 Related Links

- Documentation: 
https://medium.com/@dickareynaldisiahaan/viper-architecture-in-swift-building-robust-ios-apps-ed30d673d459
https://developer.apple.com/documentation/swiftui/navigationview

---

### 📷 Screenshots (Optional)



<img width="318" alt="Ekran Resmi 2024-12-28 20 32 02" src="https://github.com/user-attachments/assets/f687208c-b15a-4663-8af6-1e56babe67e5" />

<img width="318" alt="Ekran Resmi 2024-12-28 20 32 07" src="https://github.com/user-attachments/assets/9d821656-8741-4791-8a4e-127f11cb98d8" />

<img width="318" alt="Ekran Resmi 2024-12-28 20 32 12" src="https://github.com/user-attachments/assets/b8862343-d4bb-458d-84bb-c02c9ff17007" />
